network/discovery/options.go

```diff
@@ -91,12 +95,13 @@ func (opts *DiscV5Options) DiscV5Cfg(logger *zap.Logger) (*discover.Config, erro
 		dv5Cfg.Bootnodes = bootnodes
 	}
 
-	if logger != nil {
-		logger.Info("discovery trace is active")
-		newLogger := log.New()
-		newLogger.SetHandler(&dv5Logger{logger.Named("dv5Logger")})
-		dv5Cfg.Log = newLogger
+	newLogger := log.New()
+	if opts.EnableLogging {
```

Comment:
optimization: If EnableLogging is false, we should probably not set cfg.Log at all, so that discovery doesnt even try to log